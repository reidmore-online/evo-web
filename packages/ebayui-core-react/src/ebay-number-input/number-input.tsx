import React, { FC, useRef, useState } from "react";
import classNames from "classnames";
import { EbayTextbox, EbayTextboxPrefixIcon, EbayTextboxPostfixIcon } from "../ebay-textbox";
import type { EbayNumberInputProps } from "./types";
import { EbayIconButton } from "../ebay-icon-button";
import { EbayIconRemove24 } from "../ebay-icon/icons/ebay-icon-remove-24";
import { EbayIconDelete24 } from "../ebay-icon/icons/ebay-icon-delete-24";
import { EbayIconAdd24 } from "../ebay-icon/icons/ebay-icon-add-24";

const EbayNumberInput: FC<EbayNumberInputProps> = (props) => {
    const {
        className = "",
        value = 1,
        min = 1,
        max = Infinity,
        label = "",
        a11yDeleteText = "",
        onChange = () => {},
        onInputChange = () => {},
        onFocus = () => {},
        onBlur = () => {},
        onKeyDown = () => {},
        onKeyPress = () => {},
        onKeyUp = () => {},
        onDeleteClick = () => {},
        onIncrement = () => {},
        onDecrement = () => {},
        ...rest
    } = props;
    const defaultMin = a11yDeleteText ? 1 : 0;
    const [inputValue, setInputValue] = useState<number>(value ?? defaultMin);
    const containerRef = useRef<HTMLSpanElement>(null);

    const checkBoundary = (val, inc: number = 0): number => {
        const currentDefaultMin = a11yDeleteText ? 1 : 0;
        let newValue = parseInt(val, 10) + inc;
        if (isNaN(newValue)) {
            newValue = currentDefaultMin;
        } else if (newValue > max) {
            newValue = max;
        } else if (newValue < min) {
            newValue = min;
        }
        return newValue;
    };

    const handleAnimation = (action: string) => {
        if (!containerRef.current) return;

        const el = containerRef.current;
        el.classList.remove(`number-input--increment`);
        el.classList.remove(`number-input--decrement`);
        el.classList.remove(`number-input--increment-disabled`);
        el.classList.remove(`number-input--decrement-disabled`);

        // Trigger a reflow to ensure the animation starts
        void el.offsetWidth;

        // Add the class for the animation
        el.classList.add(`number-input--${action}`);
    };

    const handleIncrement = (e) => {
        const newValue = checkBoundary(inputValue, 1);
        handleAnimation(newValue >= max ? "increment-disabled" : "increment");

        setInputValue(newValue);
        onIncrement(e, { value: newValue });
        onChange(e, { value: newValue });
    };

    const handleDecrement = (e) => {
        const newValue = checkBoundary(inputValue, -1);
        handleAnimation(newValue <= min ? "decrement-disabled" : "decrement");

        setInputValue(newValue);
        onDecrement(e, { value: newValue });
        onChange(e, { value: newValue });
    };

    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const newValue = checkBoundary(e.target.value, 0);

        setInputValue(newValue);
        onInputChange(e, { value: newValue });
    };

    // Create wrapper functions for the event handlers
    const handleFocus = (e: React.FocusEvent<HTMLInputElement>, eventProps: { value: string }) => {
        onFocus(e, { value: Number(eventProps.value) });
    };

    const handleBlur = (e: React.FocusEvent<HTMLInputElement>, eventProps: { value: string }) => {
        onBlur(e, { value: Number(eventProps.value) });
    };

    const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>, eventProps: { value: string }) => {
        onKeyDown(e, { value: Number(eventProps.value) });
    };

    const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>, eventProps: { value: string }) => {
        onKeyPress(e, { value: Number(eventProps.value) });
    };

    const handleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>, eventProps: { value: string }) => {
        onKeyUp(e, { value: Number(eventProps.value) });
    };

    const handleDeleteClick = (e) => {
        onDeleteClick(e, { value: inputValue });
    };

    return (
        <span
            ref={containerRef}
            className={classNames(
                `number-input`,
                a11yDeleteText && inputValue === 1 && "number-input--show-delete",
                className,
            )}
        >
            <EbayTextbox
                max={max}
                min={min}
                onBlur={handleBlur}
                onChange={handleInputChange}
                onFocus={handleFocus}
                onInputChange={handleInputChange}
                onKeyDown={handleKeyDown}
                onKeyPress={handleKeyPress}
                onKeyUp={handleKeyUp}
                type="number"
                value={inputValue}
            >
                <EbayTextboxPrefixIcon>
                    {label && <label htmlFor={rest?.id || "number-input"}>{label}</label>}
                    <EbayIconButton
                        aria-hidden
                        className="number-input__decrement"
                        disabled={inputValue <= min}
                        icon={<EbayIconRemove24 />}
                        onClick={handleDecrement}
                        size="small"
                        tabIndex={-1}
                        transparent
                    />
                </EbayTextboxPrefixIcon>
                <EbayTextboxPostfixIcon>
                    {a11yDeleteText && (
                        <EbayIconButton
                            aria-label={a11yDeleteText}
                            className="number-input__delete"
                            icon={<EbayIconDelete24 />}
                            onClick={handleDeleteClick}
                            size="small"
                            transparent
                        />
                    )}
                    <EbayIconButton
                        aria-hidden
                        className="number-input__increment"
                        disabled={inputValue >= max}
                        icon={<EbayIconAdd24 />}
                        onClick={handleIncrement}
                        size="small"
                        tabIndex={-1}
                        transparent
                    />
                </EbayTextboxPostfixIcon>
            </EbayTextbox>
        </span>
    );
};

export default EbayNumberInput;
