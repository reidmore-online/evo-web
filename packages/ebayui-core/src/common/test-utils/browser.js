import { fireEvent, createEvent } from "@marko/testing-library";
/**
 * Simulates a touch based scroll event over 4 animation frames.
 *
 * @param {HTMLElement} el The element to scroll.
 * @param {number} to The new scrollLeft for the element.
 * @param {function} cb A callback to call after the scroll.
 */
function simulateScroll(el, to, cb) {
    fireEvent.scroll(el);
    el.scrollLeft = to;
    setTimeout(cb, 600);
}

function waitFrames(count, cb) {
    if (count) {
        return requestAnimationFrame(() => {
            waitFrames(count - 1, cb);
        });
    }

    cb();
}

async function pressKey(el, info) {
    for (const event of [
        createEvent.keyDown(el, info),
        createEvent.keyUp(el, info),
    ]) {
        // we assign properties to them for older browsers (chrome 49)
        Object.keys(info).forEach((key) => {
            if (event[key] !== info[key]) {
                Object.defineProperty(event, key, { value: info[key] });
            }
        });

        await fireEvent(el, event);
    }
}

const fastAnimations = {
    // Adds an style to the document which forces all transitions to run more quickly for the tests.
    start(disable) {
        if (this.fastAnimationStyle) {
            return;
        }
        const time = disable ? 0 : 1;

        this.fastAnimationStyle = document.createElement("style");
        this.fastAnimationStyle.innerHTML = `*,details::details-content,*::before,*::after {
                animation-duration: ${time}ms !important;
                transition-duration: ${time}ms !important;
                transition-delay: ${time}ms !important;
                scroll-behavior: auto !important;
            }`;
        document.head.appendChild(this.fastAnimationStyle);
    },
    stop() {
        if (!this.fastAnimationStyle) {
            return;
        }

        document.head.removeChild(this.fastAnimationStyle);
        this.fastAnimationStyle = undefined;
    },
};

export { fastAnimations, waitFrames, pressKey, simulateScroll };
