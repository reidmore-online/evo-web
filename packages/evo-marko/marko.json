{
  "taglib-id": "evo-marko",
  "tags-dir": "./src/tags",
  "taglib-imports": [
    "./src/tags/evo-icon/marko.json",
    "./src/tags/evo-skeleton/marko.json"
  ]
}
