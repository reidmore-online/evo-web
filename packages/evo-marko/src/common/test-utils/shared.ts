export function getNItems(n: number, getAttrs: (i: number) => any[]) {
  return Array.from({ length: n }).map((_, i) => getAttrs(i));
}

// Helper method to render pagination items
export function getPaginationItems(
  n: number,
  href: boolean = false,
  selected: number = -1,
  navDisabled: boolean = false,
  skipPrevNext: boolean = false,
) {
  const items = [];
  if (!skipPrevNext) {
    items.push({
      type: "previous",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  for (let i = 0; i < n; i++) {
    items.push({
      href: href ? "#" : null,
      current: i === selected,
    });
  }
  if (!skipPrevNext) {
    items.push({
      type: "next",
      href: href ? "#" : null,
      disabled: navDisabled,
    });
  }
  return items;
}
