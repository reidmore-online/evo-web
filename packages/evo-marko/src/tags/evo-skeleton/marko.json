{
  "taglib-id": "evo-marko-skeleton",
  "tags-dir": "./tags"
}
