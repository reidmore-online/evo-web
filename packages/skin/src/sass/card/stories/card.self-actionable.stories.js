export default { title: "Skin/Card/Self Actionable" };

export const verticalButton = () => `
<button class="card">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const verticalLink = () => `
<a class="card" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;

export const horizontalButton = () => `
<button class="card card--horizontal">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            Professional-grade rookie cards featuring future Hall of Famers in their awkward rookie photo phase.
        </div>
    </div>
</button>
`;

export const horizontalLink = () => `
<a class="card card--horizontal" href="https://ebay.com">
    <div class="card__media">
        <img src="/img/cards-card.jpeg" alt="Sports trading cards" class="card__hero-image">
    </div>
    <div class="card__main">
        <h3 class="card__title">Authentic Rookie Cards</h3>
        <div class="card__description">
            A collection proving that somewhere between a curveball, a three-pointer, and a slap shot lies the secret to happiness.
        </div>
    </div>
</a>
`;
