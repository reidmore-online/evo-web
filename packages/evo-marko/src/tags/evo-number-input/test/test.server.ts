import { describe, it } from "vitest";
import { composeStories } from "@storybook/marko";
import { snapshotHTML } from "../../../common/test-utils/snapshots";
import * as stories from "../number-input.stories";

const { Default, withDelete, withLabel } = composeStories(stories);

describe("evo-number-input", () => {
    it("renders default number input", async () => {
        await snapshotHTML(Default);
    });

    it("renders number input with label", async () => {
        await snapshotHTML(withLabel);
    });

    it("renders number input with delete", async () => {
        await snapshotHTML(withDelete);
    });

    it("renders number input with delete hidden", async () => {
        await snapshotHTML(withDelete, { value: "12" });
    });
});
