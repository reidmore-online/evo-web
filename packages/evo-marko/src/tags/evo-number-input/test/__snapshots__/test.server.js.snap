// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`evo-number-input > renders default number input 1`] = `
"[36m<DocumentFragment>[39m
  [36m<span[39m
    [33mclass[39m=[32m"number-input"[39m
  [36m>[39m
    [36m<span[39m
      [33mclass[39m=[32m"textbox"[39m
    [36m>[39m
      [36m<button[39m
        [33maria-hidden[39m=[32m"true"[39m
        [33mclass[39m=[32m"number-input__decrement icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mdisabled[39m=[32m""[39m
        [33mtabindex[39m=[32m"-1"[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-remove-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M21 11H3a1 1 0 0 0 0 2h18a1 1 0 0 0 0-2Z"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-remove-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
      [36m<input[39m
        [33maria-label[39m=[32m"Enter a number"[39m
        [33mclass[39m=[32m"textbox__control"[39m
        [33mid[39m=[32m"s0-0-number-input"[39m
        [33mmax[39m=[32m"100"[39m
        [33mmin[39m=[32m"0"[39m
        [33mtype[39m=[32m"number"[39m
        [33mvalue[39m=[32m"0"[39m
      [36m/>[39m
      [36m<button[39m
        [33maria-hidden[39m=[32m"true"[39m
        [33mclass[39m=[32m"number-input__increment icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mtabindex[39m=[32m"-1"[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-add-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M21 11h-8V3a1 1 0 0 0-2 0v8H3a1 1 0 0 0 0 2h8v8a1 1 0 0 0 2 0v-8h8a1 1 0 0 0 0-2Z"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-add-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
    [36m</span>[39m
  [36m</span>[39m
[36m</DocumentFragment>[39m"
`;

exports[`evo-number-input > renders number input with delete 1`] = `
"[36m<DocumentFragment>[39m
  [36m<span[39m
    [33mclass[39m=[32m"number-input number-input--show-delete"[39m
  [36m>[39m
    [36m<span[39m
      [33mclass[39m=[32m"textbox"[39m
    [36m>[39m
      [36m<button[39m
        [33maria-hidden[39m=[32m"true"[39m
        [33mclass[39m=[32m"number-input__decrement icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mdisabled[39m=[32m""[39m
        [33mtabindex[39m=[32m"-1"[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-remove-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M21 11H3a1 1 0 0 0 0 2h18a1 1 0 0 0 0-2Z"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-remove-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
      [36m<input[39m
        [33maria-label[39m=[32m"Enter a number"[39m
        [33mclass[39m=[32m"textbox__control"[39m
        [33mid[39m=[32m"s0-0-number-input"[39m
        [33mmax[39m=[32m"Infinity"[39m
        [33mmin[39m=[32m"1"[39m
        [33mtype[39m=[32m"number"[39m
        [33mvalue[39m=[32m"1"[39m
      [36m/>[39m
      [36m<button[39m
        [33maria-label[39m=[32m"Delete item"[39m
        [33mclass[39m=[32m"number-input__delete icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-delete-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M11 11a1 1 0 1 0-2 0v6a1 1 0 1 0 2 0v-6Zm3-1a1 1 0 0 1 1 1v6a1 1 0 1 1-2 0v-6a1 1 0 0 1 1-1Z"[39m
              [36m/>[39m
              [36m<path[39m
                [33mclip-rule[39m=[32m"evenodd"[39m
                [33md[39m=[32m"m7.22 5 .431-1.728A3 3 0 0 1 10.561 1h2.878a3 3 0 0 1 2.91 2.272L16.781 5H22a1 1 0 1 1 0 2h-1.59l-1.244 13.28A3 3 0 0 1 16.179 23H7.82a3 3 0 0 1-2.987-2.72L3.59 7H2a1 1 0 0 1 0-2h5.22Zm2.372-1.243a1 1 0 0 1 .97-.757h2.877a1 1 0 0 1 .97.757L14.719 5H9.282l.31-1.243ZM7.979 7h10.422l-1.227 13.093a1 1 0 0 1-.995.907H7.82a1 1 0 0 1-.995-.907L5.598 7H7.98Z"[39m
                [33mfill-rule[39m=[32m"evenodd"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-delete-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
      [36m<button[39m
        [33maria-hidden[39m=[32m"true"[39m
        [33mclass[39m=[32m"number-input__increment icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mtabindex[39m=[32m"-1"[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-add-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M21 11h-8V3a1 1 0 0 0-2 0v8H3a1 1 0 0 0 0 2h8v8a1 1 0 0 0 2 0v-8h8a1 1 0 0 0 0-2Z"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-add-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
    [36m</span>[39m
  [36m</span>[39m
[36m</DocumentFragment>[39m"
`;

exports[`evo-number-input > renders number input with delete hidden 1`] = `
"[36m<DocumentFragment>[39m
  [36m<span[39m
    [33mclass[39m=[32m"number-input"[39m
  [36m>[39m
    [36m<span[39m
      [33mclass[39m=[32m"textbox"[39m
    [36m>[39m
      [36m<button[39m
        [33maria-hidden[39m=[32m"true"[39m
        [33mclass[39m=[32m"number-input__decrement icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mtabindex[39m=[32m"-1"[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-remove-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M21 11H3a1 1 0 0 0 0 2h18a1 1 0 0 0 0-2Z"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-remove-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
      [36m<input[39m
        [33maria-label[39m=[32m"Enter a number"[39m
        [33mclass[39m=[32m"textbox__control"[39m
        [33mid[39m=[32m"s0-0-number-input"[39m
        [33mmax[39m=[32m"Infinity"[39m
        [33mmin[39m=[32m"1"[39m
        [33mtype[39m=[32m"number"[39m
        [33mvalue[39m=[32m"12"[39m
      [36m/>[39m
      [36m<button[39m
        [33maria-label[39m=[32m"Delete item"[39m
        [33mclass[39m=[32m"number-input__delete icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-delete-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M11 11a1 1 0 1 0-2 0v6a1 1 0 1 0 2 0v-6Zm3-1a1 1 0 0 1 1 1v6a1 1 0 1 1-2 0v-6a1 1 0 0 1 1-1Z"[39m
              [36m/>[39m
              [36m<path[39m
                [33mclip-rule[39m=[32m"evenodd"[39m
                [33md[39m=[32m"m7.22 5 .431-1.728A3 3 0 0 1 10.561 1h2.878a3 3 0 0 1 2.91 2.272L16.781 5H22a1 1 0 1 1 0 2h-1.59l-1.244 13.28A3 3 0 0 1 16.179 23H7.82a3 3 0 0 1-2.987-2.72L3.59 7H2a1 1 0 0 1 0-2h5.22Zm2.372-1.243a1 1 0 0 1 .97-.757h2.877a1 1 0 0 1 .97.757L14.719 5H9.282l.31-1.243ZM7.979 7h10.422l-1.227 13.093a1 1 0 0 1-.995.907H7.82a1 1 0 0 1-.995-.907L5.598 7H7.98Z"[39m
                [33mfill-rule[39m=[32m"evenodd"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-delete-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
      [36m<button[39m
        [33maria-hidden[39m=[32m"true"[39m
        [33mclass[39m=[32m"number-input__increment icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mtabindex[39m=[32m"-1"[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-add-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M21 11h-8V3a1 1 0 0 0-2 0v8H3a1 1 0 0 0 0 2h8v8a1 1 0 0 0 2 0v-8h8a1 1 0 0 0 0-2Z"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-add-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
    [36m</span>[39m
  [36m</span>[39m
[36m</DocumentFragment>[39m"
`;

exports[`evo-number-input > renders number input with label 1`] = `
"[36m<DocumentFragment>[39m
  [36m<span[39m
    [33mclass[39m=[32m"number-input"[39m
  [36m>[39m
    [36m<span[39m
      [33mclass[39m=[32m"textbox"[39m
    [36m>[39m
      [36m<label[39m
        [33mfor[39m=[32m"s0-0-number-input"[39m
      [36m>[39m
        [0mEnter a number[0m
      [36m</label>[39m
      [36m<button[39m
        [33maria-hidden[39m=[32m"true"[39m
        [33mclass[39m=[32m"number-input__decrement icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mdisabled[39m=[32m""[39m
        [33mtabindex[39m=[32m"-1"[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-remove-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M21 11H3a1 1 0 0 0 0 2h18a1 1 0 0 0 0-2Z"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-remove-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
      [36m<input[39m
        [33mclass[39m=[32m"textbox__control"[39m
        [33mid[39m=[32m"s0-0-number-input"[39m
        [33mmax[39m=[32m"100"[39m
        [33mmin[39m=[32m"0"[39m
        [33mtype[39m=[32m"number"[39m
        [33mvalue[39m=[32m"0"[39m
      [36m/>[39m
      [36m<button[39m
        [33maria-hidden[39m=[32m"true"[39m
        [33mclass[39m=[32m"number-input__increment icon-btn icon-btn--transparent icon-btn--small"[39m
        [33mdata-ebayui[39m=[32m""[39m
        [33mtabindex[39m=[32m"-1"[39m
        [33mtype[39m=[32m"button"[39m
      [36m>[39m
        [36m<svg[39m
          [33maria-hidden[39m=[32m"true"[39m
          [33mclass[39m=[32m"icon icon--24"[39m
          [33mfocusable[39m=[32m"false"[39m
        [36m>[39m
          [36m<defs>[39m
            [36m<symbol[39m
              [33mid[39m=[32m"icon-add-24"[39m
              [33mviewBox[39m=[32m"0 0 24 24"[39m
            [36m>[39m
              [36m<path[39m
                [33md[39m=[32m"M21 11h-8V3a1 1 0 0 0-2 0v8H3a1 1 0 0 0 0 2h8v8a1 1 0 0 0 2 0v-8h8a1 1 0 0 0 0-2Z"[39m
              [36m/>[39m
            [36m</symbol>[39m
          [36m</defs>[39m
          [36m<use[39m
            [33mhref[39m=[32m"#icon-add-24"[39m
          [36m/>[39m
        [36m</svg>[39m
      [36m</button>[39m
    [36m</span>[39m
  [36m</span>[39m
[36m</DocumentFragment>[39m"
`;
